CMakeLists.txt

```diff
@@ -473,3 +473,43 @@ ExternalProject_Add(aws-sdk-cpp
     -DCMAKE_INSTALL_PREFIX:PATH=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/aws-sdk-cpp
   PATCH_COMMAND python3 ${CMAKE_CURRENT_SOURCE_DIR}/tools/install_src.py --src <SOURCE_DIR> ${INSTALL_SRC_DEST_ARG}
 )
+
+#
+# Build OpenTelemetry C++ library
+# [FIXME] Remove WIN flag when windos build works
+#         and add WIN location of Protobuf
+#
+if (NOT WIN32)
+  set(PROTO_LIB_PATH "${TRITON_THIRD_PARTY_INSTALL_PREFIX}/protobuf/lib/libprotobuf.a")
+  ExternalProject_Add(opentelemetry-cpp
+    PREFIX opentelemetry-cpp
+    GIT_REPOSITORY "https://github.com/open-telemetry/opentelemetry-cpp.git"
+    GIT_TAG "v1.8.3"
+    SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/opentelemetry-cpp/src/opentelemetry-cpp"
+    EXCLUDE_FROM_ALL ON
+    CMAKE_CACHE_ARGS
+      -DProtobuf_LIBRARIES:STRING=${PROTO_LIB_PATH}
+      -DProtobuf_INCLUDE_DIR:STRING=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/protobuf/include
+      -DProtobuf_PROTOC_EXECUTABLE:STRING=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/protobuf/bin/protoc
+      -Dabsl_DIR:PATH=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/absl/lib/cmake/absl
+      -Dnlohmann_json_DIR:PATH=${TRITON_THIRD_PARTY_INSTALL_PREFIX}/nlohmann_json/lib/cmake/nlohmann_json
+      ${_CMAKE_ARGS_CMAKE_TOOLCHAIN_FILE}
+      ${_CMAKE_ARGS_VCPKG_TARGET_TRIPLET}
+      -DBUILD_TESTING:BOOL=OFF
+      -DWITH_EXAMPLES:BOOL=OFF 
+      -DWITH_BENCHMARK:BOOL=OFF
+      -DWITH_ABSEIL:BOOL=ON
+      -DWITH_OTLP:BOOL=ON
+      -DWITH_OTLP_GRPC:BOOL=OFF
```

Comment:
Is WITH_OTLP_GRPC supposed to be off? Can you add a comment about it if so? I see `grpc` as a dependency.

Reply:
Yes for now, `grpc` is a dependency because `-DWITH_OTLP=ON`, I ran into issues when `grpc` was not installed prior to `opentelemetry` build with  `-DWITH_OTLP=ON`. Currently `-DWITH_OTLP_GRPC:BOOL=OFF`, because OTLP?gRPC exporter is not supported, due to conflicts raised during build. After proper debugging, `-DWITH_OTLP_GRPC:BOOL` will be set to `ON` and OTLP/gRPC exporters will be supported

Reply:
Sure, feel free to file smaller tickets for follow-up tasks. You can group them under an OTLP or Tracing epic or something if not already